### PR TITLE
Added docs for Config and Auth & update doc workflow

### LINKED ISSUE #429: Document required Authentication steps in codeflare-sdk and ClusterConfiguration Parameters
### Name of Feature or Improvement

- Document the necessary steps for authenticating Codeflare-sdk during cluster.up() 
- Its good to have brief description about ClusterConfiguration Parameters.     
        For example Namespace should be created , which is configuring in ClusterConfiguration Parameters and about other parameters necessary

### PR DESCRIPTION
**Pull request by Bobbins228**
# Issue link

Closes #429
# What changes have been made

The detailed "code orientated" docs have been moved to a new folder within docs called `detailed-docs`
The release workflow has been updated to account for this.
New docs were added for Cluster Configuration and Authentication

Ingress instructions in README moved to cluster-configuration.md
# Verification steps

You can verify the workflow changes by running this command.
```
poetry run pdoc --html -o docs/detailed-documentation src/codeflare_sdk && pushd docs/detailed-documentation && rm -rf cluster job utils && mv codeflare_sdk/* . && rm -rf codeflare_sdk && popd && find docs/detailed-documentation -type f -name "*.html" -exec bash -c "echo '' >> {}" \;
``` 
You should get a docs/detailed-docs folder containing the detailed documentation.
## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [x] Testing is not required for this change

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Method 1 Token Authentication
+This is how a typical user would authenticate to their cluster using `TokenAuthentication`.
+```
+from codeflare_sdk.cluster.auth import TokenAuthentication
+
+auth = TokenAuthentication(
+    token = "XXXXX",
+    server = "XXXXX",
+    skip_tls=False,
+    # ca_cert_path="/path/to/cert
```
Comment (VanillaSpoon) [inline docs/authentication.md:14]: Just a small nit, but other than that LGTM, great job @Bobbins228 :)
```diff
- # ca_cert_path="/path/to/cert
+ # ca_cert_path="/path/to/cert"
```

**Review by VanillaSpoon — COMMENTED** [review]
Tiny nit, but other than that lgtm, great job @Bobbins228 :)

**Review by VanillaSpoon — COMMENTED** [review]
lgtm :)

File: docs/detailed-documentation/cluster/config.html [hunk 2]
```diff
@@ -100,7 +100,7 @@ <h2 class="section-title" id="header-classes">Classes</h2>
...
+<span>(</span><span>name: str, namespace: str = None, head_info: list = &lt;factory&gt;, head_cpus: int = 2, head_memory: int = 8, head_gpus: int = 0, machine_types: list = &lt;factory&gt;, min_cpus: int = 1, max_cpus: int = 1, num_workers: int = 1, min_memory: int = 2, max_memory: int = 2, num_gpus: int = 0, template: str = '/home/mark/distributed-workloads-repos/codeflare-sdk/src/codeflare_sdk/templates/base-template.yaml', instascale: bool = False, mcad: bool = True, envs: dict = &lt;factory&gt;, image: str = '', local_interactive: bool = False, image_pull_secrets: list = &lt;factory&gt;, dispatch_priority: str = None, openshift_oauth: bool = False, ingress_options: dict = &lt;factory&gt;, ingress_domain: str = None)</span>
```
Comment (ChristianZaccaria) [inline docs/detailed-documentation/cluster/config.html:103]: This should be reverted to its original path. This must've been generated when running the mentioned poetry command.

File: docs/detailed-documentation/job/jobs.html [hunk 3]
```diff
@@ -323,7 +323,7 @@ <h3>Methods</h3>
...
-<span>(</span><span>script: Optional[str] = None, m: Optional[str] = None, script_args: Optional[List[str]] = None, name: Optional[str] = None, cpu: Optional[int] = None, gpu: Optional[int] = None, memMB: Optional[int] = None, h: Optional[str] = None, j: Optional[str] = None, env: Optional[Dict[str, str]] = None, max_retries: int = 0, mounts: Optional[List[str]] = None, rdzv_port: int = 29500, rdzv_backend: str = None, scheduler_args: Optional[Dict[str, str]] = None, image: Optional[str] = None, workspace: Optional[str] = 'file:///home/runner/work/codeflare-sdk/codeflare-sdk')</span>
+<span>(</span><span>script: Optional[str] = None, m: Optional[str] = None, script_args: Optional[List[str]] = None, name: Optional[str] = None, cpu: Optional[int] = None, gpu: Optional[int] = None, memMB: Optional[int] = None, h: Optional[str] = None, j: Optional[str] = None, env: Optional[Dict[str, str]] = None, max_retries: int = 0, mounts: Optional[List[str]] = None, rdzv_port: int = 29500, rdzv_backend: str = None, scheduler_args: Optional[Dict[str, str]] = None, image: Optional[str] = None, workspace: Optional[str] = 'file:///home/mark/distributed-workloads-repos/codeflare-sdk')</span>
```
Comment (ChristianZaccaria) [inline docs/detailed-documentation/job/jobs.html:326]: Here too. This should be reverted to its original path.

**Review by ChristianZaccaria — CHANGES_REQUESTED** [review]
Looks great! I left a couple nitpicks.

**Review by Fiona-Waters — COMMENTED** [review]
This is great!
Just one comment. I think we should add a `section intro` to the [jobs.html file](https://github.com/project-codeflare/codeflare-sdk/pull/435/files#diff-ce5c4b9fb85fbacb4c8b5c5436b8f13fdc55da7d9b4fdbcac88d0fc2df2d2becR24), so that all sub_modules have one. I think the ... on this page looks like something is missing, especially with the whitespace underneath. 
image
 What would you think about making these into links so it's clearer where they are coming from? Just a thought.

**Comment by Bobbins228** [comment]
@Fiona-Waters 
These html pages are generated off of comments in the code and so the modules jobs, cluster and utils would not generate any notes because they are folders within the SDK. There might be a way to add a description for these modules so that they are generated automatically but I think it would warrant a separate issue WDYT?

**Comment by Bobbins228** [comment]
My bad I thought you meant the module yeah I can add a short desc of the DDP Jobs :+1:

**Pull request merged**